tests/queries/0_stateless/03231_create_with_clone_as.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+# Tags: no-replicated-database
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS foo_memory"
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS clone_as_foo_memory"
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS foo_file"
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS clone_as_foo_file"
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS foo_merge_tree"
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS clone_as_foo_merge_tree"
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS foo_replacing_merge_tree"
+${CLICKHOUSE_CLIENT} -q "DROP TABLE IF EXISTS clone_as_foo_replacing_merge_tree"
+
+# CLONE AS with a table of Memory engine
+${CLICKHOUSE_CLIENT} --optimize_throw_if_noop 1 -q "CREATE TABLE foo_memory (x Int8, y String) ENGINE=Memory"
+${CLICKHOUSE_CLIENT} --optimize_throw_if_noop 1 -q "SHOW CREATE TABLE foo_memory"
+${CLICKHOUSE_CLIENT} --optimize_throw_if_noop 1 -q "INSERT INTO foo_memory VALUES (1, 'a'), (2, 'b')"
+
+echo "$(${CLICKHOUSE_CLIENT} --optimize_throw_if_noop 1 --server_logs_file=/dev/null -q "CREATE TABLE clone_as_foo_memory CLONE AS foo_memory" 2>&1)" \
```

Comment:
JFYI It's possible to check errors in .sql tests like this: https://github.com/ClickHouse/ClickHouse/blob/f93ccfa800330a81468e317db7f009c2c0a0290c/tests/queries/0_stateless/01275_extract_groups_check.sql#L1

Reply:
Initially, I put the test in a .sh file because I had to check the server error, I didn't know this way to check.
Now, I need to user `$CLICKHOUSE_DATABASE` to create a Replicated database, so I will stick with .sh file.

Reply:
I seem to be able to use `{database}` as a placeholder in .sql

```
ReplicatedMergeTree('/clickhouse/tables/{database}/test_00062/
```